lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobinSpec.hs

```diff
@@ -1360,18 +1476,29 @@ linearCost SelectionSkeleton{inputsSkeleton, outputsSkeleton, changeSkeleton}
     + F.length outputsSkeleton
     + F.length changeSkeleton
 
-data MakeChangeData = MakeChangeData
-    { inputBundles
-        :: NonEmpty TokenBundle
-    , extraInputCoin
-        :: Maybe Coin
-    , outputBundles
-        :: NonEmpty TokenBundle
-    , cost
-        :: Coin
-    , minCoinValueDef
-        :: MinCoinValueFor
-    } deriving (Eq, Show)
+type MakeChangeData = MakeChangeCriteria MinCoinValueFor BundleSizeAssessor
+
+data BundleSizeAssessor
+    = NoBundleSizeLimit
+      -- ^ Indicates that there is no limit on a token bundle's size.
+    | BundleAssetCountUpperLimit Int
+      -- ^ Indicates an inclusive upper bound on the number of assets in a
+      -- token bundle.
+    deriving (Eq, Show)
+
+mkBundleSizeAssessor
+    :: BundleSizeAssessor
+    -> (TokenBundle -> TokenBundleSizeAssessment)
+mkBundleSizeAssessor = \case
+    NoBundleSizeLimit ->
+        const TokenBundleSizeWithinLimit
+    BundleAssetCountUpperLimit upperLimit ->
+        \bundle ->
+            let assetCount = Set.size $ TokenBundle.getAssets bundle in
```

Comment:
Also as previously said: 

We use `Set.size` to measure the size here in the unit tests, but the CBOR-encoded length in reality.

I fear there's no coverage for problems like what 4c36492 fixes — and that we may some day could end up generating too large bundles in a performSelection corner case.

But simply documenting what we expect in the argument to performSelection should probably be good enough.

Reply:
I've also added some further property tests in 7f0059034d661b997933f1dc043f734d36a26c4e.